test/helpers/dotPathToHash.test.js

```diff
@@ -96,6 +96,18 @@ describe('dotPathToHash helper function', () => {
     done()
   })
 
+  it('detects duplicate keys with the same value when warnOnDuplicate is true', (done) => {
+    const { target, duplicate, conflict } = dotPathToHash(
+      { keyWithNamespace: 'one.two.three' },
+      { one: { two: { three: '' } } },
+      { warnOnDuplicate: true }
```

Comment:
The option is `warnOnKeyDuplicate` and not `warnOnDuplicate`. There is something flawed with the logic in this PR. This option shouldn't be needed to be passed to `dotPathToHash`. It should already return duplicates and conflicts without any extra option. This test is basically a duplicate of the one right above it.

Reply:
Addressed in your other comment below

Reply:
@andrewRichardson Is there a reason you left the test? It doesn't seem to test anything since you've removed `warnOnDuplicate`.

Reply:
Ah, you are correct.  This was left by mistake, I just pushed a commit to remove it 👍🏼